File: java/src/main/java/io/fivexer/sdk/model/WorkerDetail.java
package io.fivexer.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * Full detail of one worker, including current load.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerDetail {
    private String id;
    private List<String> tags;
    private Map<String, Double> routingWeights;
    // Which entries of routingWeights the learning layer owns, and when it last wrote them — the
    // difference between "an operator vetoed this tag" and "the model did", which is the
    // difference between a decision to keep and one to revert.
    private Map<String, Double> learnedRoutingWeights;
    // What routingWeights held before the last sync; restored by learning().revertWeights()
    private Map<String, Double> routingWeightsSnapshot;
    private Long learnedRoutingWeightsSyncedAt;
    private List<WorkerSkill> skills;  // null when skills are not in use for this worker
    private List<WorkerTeam> teams;    // null when this worker is in no teams
    private Integer maxBacklogSize;  // null when no per-worker cap is set; 0 means receive nothing
    private boolean available;
    // Why they are unavailable when it is not simply "off shift": an invite nobody accepted, or
    // a QR join waiting on operator approval. An operator UI that renders either as "paused"
    // tells the wrong story — nobody has to resume an invite, they have to chase it.
    private boolean invitePending;
    private boolean pendingApproval;
    private int queueDepth;
    /** Null when they have never chosen one and the workspace default applies. */
    private String locale;

    public String getId() { return id; }
    public List<String> getTags() { return tags; }
    public Map<String, Double> getRoutingWeights() { return routingWeights; }
    public Map<String, Double> getLearnedRoutingWeights() { return learnedRoutingWeights; }
    public Map<String, Double> getRoutingWeightsSnapshot() { return routingWeightsSnapshot; }
    public Long getLearnedRoutingWeightsSyncedAt() { return learnedRoutingWeightsSyncedAt; }
    public List<WorkerSkill> getSkills() { return skills; }
    public List<WorkerTeam> getTeams() { return teams; }
    public Integer getMaxBacklogSize() { return maxBacklogSize; }
    public boolean isAvailable() { return available; }
    public boolean isInvitePending() { return invitePending; }
    public boolean isPendingApproval() { return pendingApproval; }
    public int getQueueDepth() { return queueDepth; }

    /** The worker's language — what their push notifications and emails are composed in. */
    public String getLocale() { return locale; }
}

File: php/src/Model/WorkerDetail.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Full detail of one worker, including current load.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerDetail
{
    public function __construct(
        public readonly string $id,
        /** @var list<string> */
        public readonly array $tags,
        /** @var array<string, float>|null */
        public readonly ?array $routingWeights,
        /** @var list<WorkerSkill>|null */
        public readonly ?array $skills,
        /** 0 means receive nothing; null means no per-worker cap */
        public readonly ?int $maxBacklogSize,
        public readonly bool $available,
        public readonly int $queueDepth,
        /**
         * Which entries of $routingWeights the learning layer owns, and when it last wrote them
         * — the difference between "an operator vetoed this tag" and "the model did", which is
         * the difference between a decision to keep and one to revert.
         *
         * @var array<string, float>|null
         */
        public readonly ?array $learnedRoutingWeights = null,
        /**
         * What $routingWeights held before the last sync; restored by revertWeights().
         *
         * @var array<string, float>|null
         */
        public readonly ?array $routingWeightsSnapshot = null,
        public readonly ?int $learnedRoutingWeightsSyncedAt = null,
        /** @var list<WorkerTeam>|null */
        public readonly ?array $teams = null,
        /**
         * Why they are unavailable when it is not simply "off shift": an invite nobody accepted,
         * or a QR join waiting on operator approval. An operator UI that renders either as
         * "paused" tells the wrong story — nobody has to resume an invite, they have to chase it.
         */
        public readonly bool $invitePending = false,
        public readonly bool $pendingApproval = false,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            tags: \array_map('strval', $data['tags'] ?? []),
            routingWeights: $data['routingWeights'] ?? null,
            skills: Json::parseEachOrNull($data, 'skills', [WorkerSkill::class, 'fromArray']),
            maxBacklogSize: isset($data['maxBacklogSize']) ? (int) $data['maxBacklogSize'] : null,
            available: (bool) ($data['available'] ?? false),
            queueDepth: (int) ($data['queueDepth'] ?? 0),
            learnedRoutingWeights: $data['learnedRoutingWeights'] ?? null,
            routingWeightsSnapshot: $data['routingWeightsSnapshot'] ?? null,
            learnedRoutingWeightsSyncedAt: isset($data['learnedRoutingWeightsSyncedAt'])
                ? (int) $data['learnedRoutingWeightsSyncedAt']
                : null,
            teams: Json::parseEachOrNull($data, 'teams', [WorkerTeam::class, 'fromArray']),
            invitePending: (bool) ($data['invitePending'] ?? false),
            pendingApproval: (bool) ($data['pendingApproval'] ?? false),
        );
    }
}
